packages/teleterm/src/ui/DocumentGateway/DocumentGateway.tsx

```diff
@@ -45,8 +57,19 @@ export function DocumentGateway(props: State) {
     disconnect,
     reconnect,
     runCliCommand,
+    changeDbName,
+    changeDbNameAttempt,
+    doc,
   } = props;
 
+  const handleChangeDbName = useMemo(() => {
+    return debounce((value: string) => {
+      changeDbName(value);
+    }, 150);
+  }, []);
```

Comment:
Changing the db name fails for me ("Could not change the database name: Could not find gateway undefined") until I change some code and cause React to reload the components.

My bet is that `useMemo` memoizes `changeDbName` when `doc.gatewayUri` is still undefined (it gets updated after the gateway is actually created).

To fix this, this `useMemo` would need to depend on `changeDbName` and `changeDbName` would need to have a stable identity itself.

Reply:
It didn't happen to me, but right, it can be caused by the `useMemo`.

I updated `useAsync` too.